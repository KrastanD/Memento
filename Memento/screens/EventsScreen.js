import React, { useEffect, useState } from "react";
import { StyleSheet, Text, View, TouchableOpacity } from "react-native";
import getEventsJson from "../server/eventsAxios";
import { ScrollView } from "react-native-gesture-handler";
import Event from "../models/Event";
import Icon from "react-native-vector-icons/FontAwesome";
import { TextInput } from "react-native-paper";
import ExpandableCard from "../components/ExpandableCard";
const axios = require("axios");

export default function EventsScreen({ navigation, route }) {
  const [events, setEvents] = useState([]);
  const [dates, setDates] = useState([]);
  const [searchTerm, setSearchTerm] = useState("");

  useEffect(() => {
    const CancelToken = axios.CancelToken;
    const source = CancelToken.source();
    getEventsJson(source)
      .then((resp) => {
        var events = [];
        resp.forEach((element) => {
          events.push(new Event(element));
        });
        events = events.filter(function (event) {
          return (
            event.eventTitle.toLowerCase().indexOf(searchTerm.toLowerCase()) !=
              -1 ||
            event.description.toLowerCase().indexOf(searchTerm.toLowerCase()) !=
              -1
          );
        });
        var dict = {};
        events.forEach((item) => {
          if (item.timePostedString in dict) {
            dict[item.timePostedString].push(item);
          } else {
            dict[item.timePostedString] = [item];
          }
        });
        setEvents(dict);
        var dateArr = Object.keys(dict);
        var dateArr = dateArr.sort(
          (a, b) => dict[b][0].timePostedDate - dict[a][0].timePostedDate
        );
        setDates(dateArr);
      })
      .catch((error) => {
        if (axios.isCancel(error)) {
        } else {
          console.log(error);
        }
      });

    return function cleanup() {
      source.cancel();
    };
  }, [searchTerm, events]);

  return (
    <View style={{ flex: 1 }}>
      <React.Fragment>
        <ScrollView style={styles.container}>
          <View style={{ paddingHorizontal: 10, flexDirection: "row" }}>
            <TextInput
              label="Search"
              mode="outlined"
              theme={{
                colors: { primary: "#01a699" },
              }}
              style={{ flex: 1 }}
              onChangeText={(text) => setSearchTerm(text)}
            />
          </View>
          {events != undefined &&
            dates != undefined &&
            dates.map(
              (date) =>
                events[date] != undefined &&
                events[date]
                  .slice(0)
                  .reverse()
                  .map((event, i) => (
                    <View key={i}>
                      {event == events[date][events[date].length - 1] ? (
                        <Text
                          style={{
                            fontSize: 14,
                            color: "gray",
                            paddingTop: 15,
                            paddingLeft: 15,
                          }}
                        >
                          {date}
                        </Text>
                      ) : null}
                      <ExpandableCard key={event.id} event={event} />
                    </View>
                  ))
            )}
        </ScrollView>
        <TouchableOpacity
          style={styles.addEventStyle}
          onPress={() => navigation.navigate("AddEvent")}
        >
          <Icon name="plus" size={30} color="white" />
        </TouchableOpacity>
      </React.Fragment>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: "#fafafa",
  },
  addEventStyle: {
    borderWidth: 1,
    borderColor: "rgba(0,0,0,0.2)",
    alignItems: "center",
    justifyContent: "center",
    width: 60,
    position: "absolute",
    bottom: 20,
    right: 20,
    height: 60,
    backgroundColor: "#01a699",
    borderRadius: 120,
    elevation: 2,
  },
});
