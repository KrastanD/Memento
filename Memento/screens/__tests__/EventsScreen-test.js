import React from "react";
import EventsScreen from "../EventsScreen";
import renderer from "react-test-renderer";

it("snapshots EventsScreen screen correctly", () => {
  const tree = renderer.create(<EventsScreen />);
  expect(tree.toJSON()).toMatchSnapshot();
});
