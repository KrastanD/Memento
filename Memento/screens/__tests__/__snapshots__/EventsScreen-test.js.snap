// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots EventsScreen screen correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    collapsable={false}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    style={
      Object {
        "backgroundColor": "#fafafa",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingHorizontal": 10,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View>
            <View
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                  },
                  Object {
                    "backgroundColor": "#f6f6f6",
                    "borderColor": "rgba(0, 0, 0, 0.54)",
                    "borderRadius": 4,
                    "borderWidth": 1,
                  },
                ]
              }
            />
            <View
              style={
                Object {
                  "minHeight": 64,
                  "paddingBottom": 0,
                  "paddingTop": 8,
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#f6f6f6",
                      "bottom": 0,
                      "height": 2,
                      "left": 10,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 6,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                      ],
                      "width": 8,
                    }
                  }
                />
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "backgroundColor": "#f6f6f6",
                      "color": "transparent",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 18,
                      "opacity": 0,
                      "paddingHorizontal": 0,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 37,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "scale": 1,
                        },
                        Object {
                          "scaleY": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                >
                  Search
                </Text>
                <Text
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    Object {
                      "color": "#01a699",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 0,
                      "opacity": 0,
                      "paddingHorizontal": 14,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 36,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "scale": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                >
                  Search
                </Text>
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "rgba(0, 0, 0, 0.54)",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 0,
                      "opacity": 1,
                      "paddingHorizontal": 14,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 36,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "scale": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                >
                  Search
                </Text>
              </View>
              <TextInput
                allowFontScaling={true}
                editable={true}
                multiline={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder=""
                placeholderTextColor="rgba(0, 0, 0, 0.54)"
                rejectResponderTermination={true}
                selectionColor="#01a699"
                style={
                  Array [
                    Object {
                      "flexGrow": 1,
                      "margin": 0,
                      "paddingHorizontal": 14,
                      "textAlign": "left",
                      "zIndex": 1,
                    },
                    Object {
                      "height": 56,
                    },
                    Object {
                      "paddingBottom": 0,
                      "paddingTop": 0,
                    },
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "textAlignVertical": "center",
                    },
                  ]
                }
                underlineColorAndroid="transparent"
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#01a699",
        "borderColor": "rgba(0,0,0,0.2)",
        "borderRadius": 120,
        "borderWidth": 1,
        "bottom": 20,
        "elevation": 2,
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 20,
        "width": 60,
      }
    }
  >
    <Text />
  </View>
</View>
`;
